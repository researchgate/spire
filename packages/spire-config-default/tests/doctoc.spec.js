const { join } = require('path');
const { readFile } = require('fs-extra');
const { createFixture } = require('spire-test-utils');

const configWithDoctocPlugin = JSON.stringify({
  name: 'spire-config-default-test-doctoc',
  spire: {
    plugins: [
      'spire-config-default/doctoc',
      'spire-config-default/lint-staged',
    ],
  },
});

describe('spire-config-default/doctoc', () => {
  it('adds markdown linter', async () => {
    const fixture = await createFixture({
      'package.json': configWithDoctocPlugin,
      'readme.md': `
# First
## Second
### Third
      `,
    });
    await fixture.run('git', ['init']);
    await fixture.run('git', ['add', 'readme.md']);
    const { stdout } = await fixture.run('spire', [
      'hook',
      'precommit',
      '--debug',
    ]);
    expect(stdout).toMatch(/Running spire-config-default\/doctoc\.precommit/);
    expect(
      await readFile(join(fixture.cwd, 'readme.md'), 'utf8')
    ).toMatchSnapshot();
    await fixture.clean();
  });
});
