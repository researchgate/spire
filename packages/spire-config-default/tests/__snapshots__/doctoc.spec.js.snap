// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`spire-config-default/doctoc adds markdown linter 1`] = `
"<!-- START doctoc generated TOC please keep comment here to allow auto update -->
<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->


- [First](#first)
  - [Second](#second)
    - [Third](#third)

<!-- END doctoc generated TOC please keep comment here to allow auto update -->


# First
## Second
### Third
      "
`;
